### PR TITLE
check_maintainers: list maintainers of a patchset

### PR DESCRIPTION
**Pull request by msune**
This patch adds the functionality to check_maintainers.py to list
the maintainers of the files that a commit or a series of commits
affect.

It supports:

* Checking a (single) patch file
* Checking last N commits in the current branch
* Checking against the current HEAD with a ref (e.g. a git tag)

Other improvements:

* Adding help option '-h'
* Minor cosmetics to the code

### CODE REVIEW

**Comment by msune** [comment]
@vmaffione @sandervrijders since you were providing comments to the tool, could you please have a look?

Thanks

File: check_maintainers.py [hunk 1]
```diff
@@ -101,14 +105,85 @@ def scan_paths(maintainers, paths):
...
+			errormsg("Unable to retrieve files for the last '"+opt+"' commits. Error"+str(e))
+			sys.exit(1)
+
+	for f_ in p.stdout.read().split():
+		ms, n = get_maintainers(maintainers, f_)
+		if not check_extension(f_):
+			continue
+		if n == 0:
+			continue
+		print "F: "+f_
```
Comment (sandervrijders) [inline check_maintainers.py]: This breaks with Python 3. Correct syntax is print("F: "+f_)

**Comment by msune** [comment]
Thanks Sander. Yes I forgot to test it against Python 3 (again!, wtf). @vmaffione  has sent a PR to my branch to fix it here:

https://github.com/bisdn/irati/pull/30

Just waiting for an ammend in the commit message.

**Comment by vmaffione** [comment]
ack

**Pull request merged**